### DISCUSSION TITLE
failed to create volume - GKE w/Local SSD scratch volume

### DISCUSSION THREAD

**Discussion by jeremyrcampb** [concourse/concourse#7072] [Help & Support]
I'm working on a new 7.2.0 deployment to GKE using the helm chart.  I'm attempting to use a [new feature](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/local-ssd#creating_a_node_pool_using_ephemeral_storage_on_local_ssds) in GKE 1.18 that presents a LocalSSD to the node worker as an ephemeral volume, which is then used for workloads scratch/emptyDir/root volume.  My intention is to use this for the concourse-workers to provide SSD performance without paying a premium for persistent SSDs.

The LocalSSD Volume seems to come to the concourse-worker fine:

```
Filesystem      Size  Used Avail Use% Mounted on
overlay         369G  2.3G  348G   1% /
...
```

However, tasks error:

`run check: find or create container on worker concourse-worker-7989f54c46-sk7vj: failed to create volume`

and I get baggageClaim errors:

```
{"timestamp":"2021-05-25T18:26:05.172550651Z","level":"info","source":"baggageclaim","message":"baggageclaim.api.volume-server.get-volume.get-volume.volume-not-found","data":{"session":"3.1.64913.1","volume":"a968d98d-08c6-4c1a-6053-3ed301ca4e78"}}
{"timestamp":"2021-05-25T18:26:05.172583004Z","level":"info","source":"baggageclaim","message":"baggageclaim.api.volume-server.get-volume.volume-not-found","data":{"session":"3.1.64913","volume":"a968d98d-08c6-4c1a-6053-3ed301ca4e78"}}
{"timestamp":"2021-05-25T18:26:05.185823255Z","level":"error","source":"baggageclaim","message":"baggageclaim.api.volume-server.create-volume-async.create-volume.failed-to-materialize-strategy","data":{"error":"invalid argument","handle":"a968d98d-08c6-4c1a-6053-3ed301ca4e78","session":"3.1.64914.1"}}
{"timestamp":"2021-05-25T18:26:05.185861968Z","level":"error","source":"baggageclaim","message":"baggageclaim.api.volume-server.create-volume-async.failed-to-create","data":{"error":"invalid argument","handle":"a968d98d-08c6-4c1a-6053-3ed301ca4e78","privileged":true,"session":"3.1.64914","strategy":{"type":"cow","volume":"c8c051f4-1b49-4b07-56de-cadd6def1be2"}}}
```

Worker kind is `Deployment`, runtime is `containerd`, and baggageclaim driver is `overlay`.

Any suggestions on how to troubleshoot this?

**Comment by chenbh** [comment]
I believe you can't mount another overlay filesystem (worker) on top of an existing overlay fs (LocalSSD). So you'll have to setup the LocalSSD with some other fs or use another baggageclaim driver
